src/main/kotlin/carracing/game/domain/CarRacingModel.kt

```diff
@@ -0,0 +1,64 @@
+package carracing.game.domain
+
+import carracing.game.domain.data.Car
+import carracing.game.domain.data.Race
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlin.random.Random
+
+class CarRacingModel(
+    private val random: Random = Random,
+) {
+    private var carsAmount: Int? = null
+    private var roundsAmount: Int? = null
+
+    fun assignCarsAmount(input: String?) {
+        val inputInt = input?.toIntOrNull()
+        if (inputInt == null || inputInt < MIN_CARS_AMOUNT) {
+            throw IllegalArgumentException("Cars amount should be at least $MIN_CARS_AMOUNT")
+        }
```

Comment:
`require()` throws an `IllegalArgumentException` if the given condition is not met.